File: src/main/java/com/example/Employee.java
package com.example;

import com.fasterxml.jackson.annotation.JsonMerge;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;
import java.util.Objects;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Employee implements Comparable<Employee> {
    private long id;
    private String name;
    private int wage;
    private String position;
    @JsonMerge
    private Map<Long, Employee> colleagues;

    @Override
    public int compareTo(Employee employee) {
        return name.compareTo(employee.name);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Employee employee = (Employee) o;
        return id == employee.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/kotlin/com/example/kotlin/Employee.kt
package com.example.kotlin

import com.fasterxml.jackson.annotation.JsonMerge
import jakarta.validation.Valid
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern

data class Employee(
    @field:Min(0)
    var id: Long = -1,

    @field:NotBlank
    @field:Pattern(regexp = "^[\\w'\\-,.][^0-9_!¡?÷¿/\\\\+=@#$%ˆ&*(){}|~<>;:\\[\\]]{2,}$")
    var name: String = "",

    @field:Min(500)
    @field:Max(5000)
    var wage: Int = 0,

    @field:NotBlank
    var position: String = "",

    @JsonMerge
    @field:Valid
    var colleagues: Map<Long, Employee> = emptyMap()
)
